src/NServiceBus.Core/SettingsExtensions.cs

```diff
@@ -34,7 +34,7 @@ public static LogicalAddress LogicalAddress(this ReadOnlySettings settings)
         {
             Guard.AgainstNull(nameof(settings), settings);
 
-            if (!settings.TryGet<ReceiveConfiguration>(out var receiveConfiguration))
+            if (!settings.TryGet<ReceiveComponent.Configuration>(out var receiveConfiguration))
```

Comment:
isn't it weird that we're using the configuration here instead of the settings?

Reply:
Talked to @timbussmann and we concluded that this us ugly but what the current code is doing. We might be able to fix this in a follow up PR